UI & branding improvements, notifications, manual dark mode, and more message types

micahlt opened the pull request:
This is possibly the biggest block of changes since Itchy moved from Cordova to Capacitor.  I've tried to organize my commits as much as I can, so it should be relatively easy to review.  Here is a zipped APK file for testing: [0-9-1-prerelease.zip](https://github.com/Scratch-Client-4/itchy-ionic/files/7696099/0-9-1-prerelease.zip)

**Breaking changes:** 

- Upgraded to Android Studio Arctic Fox (latest)
- Upgraded to Ionic 5
- Upgraded to Capacitor 3.x
- Updated to Gradle 7

**User-facing changes:**

- More supported message types
- More supported feed types
- New (non-copyright-infringing) branding
- Share button for projects
- Local notifications every 30 minutes
- "Projects Friends Loved" section on Explore
- Much better user page
- Dynamic status bar colors to match dark mode

**Backend changes:**

- Breaking changes listed above
- Addition of `Share`, `App`, and `LocalNotifications` Capacitor plugins
- More requests done locally rather than relying on `itchy-api`
- Fixed back button behavior in modals
- Fixed manual dark mode toggle (close #26)
- Moved to ScratchDB v3 by DatOneLefty

locness3 commented:
> "Projects Friends Loved" section on Explore

Does that mean followed people?

> Much better user page

And broken "Open your profile" link in Settings.
Also, opening your profile from the settings, then closing it, then going to Explore will open it again

> Dynamic status bar colors to match dark mode

Both my status bar and navigation bar are black on light mode. Is this intended? (android 9)

locness3 commented on src/components/Message.vue:
```diff
@@ -27,15 +27,21 @@
-    <ion-note v-if="m.type == 'remixproject'">
-      <ion-icon :icon="colorPalette" class="blue"></ion-icon> remixed your project {{ m.parent_title }}
+    <ion-note v-if="m.type == 'curatorinvite'">
+      <ion-icon :icon="images" class="blue"></ion-icon> wants you to curate {{ m.title }}
```
*invited you

micahlt replied:
What is the reason for this change?

locness3 replied:
Makes it consistent with the website.

micahlt replied:
Fair enough.  Will fix.

locness3 commented on src/components/Message.vue:
```diff
@@ -231,34 +290,29 @@ export default defineComponent({
         console.error(res.statusText);
       }
     },
-    async thankUser(o) {
```
thank??

micahlt replied:
Essentially this leads the current user to the following user's profile comments to thank them for the follow.

locness3 replied:
I get it - but the function name is not correct here.

micahlt replied:
What would you like me to rename it?

locness3 commented on src/components/ProjectModal.vue:
```diff
@@ -169,10 +184,21 @@ export default defineComponent({
+    async shareProject() {
+      await Share.share({
+        title: this.title,
+        text: 'Check out this Scratch project I found with Itchy!',
```
IMO, this text is pretty unnecessary. I personnally dislike when apps add text to shared links.

micahlt replied:
Agreed.  I'll fix that this afternoon.

locness3 commented on src/views/SettingsPage.vue:
```diff
@@ -43,7 +43,7 @@
       <ion-item>
         <ion-label>
-          Enable personal feed
+          Enable personal data on Explore
```
Enable personal feed on Explore.

micahlt replied:
> Enable personal feed on Explore

I would have used this, but it also uses personal data for the "Projects Friends Loved" section.

micahlt commented:
> > "Projects Friends Loved" section on Explore
> 
> Does that mean followed people?
> 
> > Much better user page
> 
> And broken "Open your profile" link in Settings.
> Also, opening your profile from the settings, then closing it, then going to Explore will open it again
> 
> > Dynamic status bar colors to match dark mode
> 
> Both my status bar and navigation bar are black on light mode. Is this intended? (android 9)


Yes, "Friends" refers to followed users, just like the Scratch 1.x era.  I can fix the "open your profile" link this afternoon.  As far as routing issues, that was an existing issue that will be resolved in a future PR.  What color is your light mode status bar normally?  And imo, does it look bad to you?

locness3 commented:
> What color is your light mode status bar normally?

It always was black AFAIRC, but I assume it should be white on light mode?

> Yes, "Friends" refers to followed users, just like the Scratch 1.x era.
I don't think people who would use this app would be aware - should change this.

locness3 commented:
![Screenshot_20211211-151329](https://user-images.githubusercontent.com/37651007/145680015-ec2f692d-b94b-4066-a31b-a0990c91d1e0.png)
Also, I don't have this much messages.

micahlt commented:
> > What color is your light mode status bar normally?
> 
> It always was black AFAIRC, but I assume it should be white on light mode?
> 
> > Yes, "Friends" refers to followed users, just like the Scratch 1.x era.
> 
> I don't think people who would use this app would be aware - should change this.

Most of the apps I use continue to use a dark status bar even in light mode - I don't think that's an issue.  As far as renaming that section of Explore, I'm not sure.  I think it should be pretty logical since it's already the same layout of the Scratch homepage.  The main reason I used the current phrase was to keep the line length short.  Anything longer than three mid-size words creates overflow which is unacceptable.  Any suggestions?

micahlt commented:
> ![Screenshot_20211211-151329](https://user-images.githubusercontent.com/37651007/145680015-ec2f692d-b94b-4066-a31b-a0990c91d1e0.png)
> Also, I don't have this much messages.

Very odd - can you debug with chrome devtools?

micahlt commented:
I just pushed a fix for the function names and strings (a6987d5).  The messages issue is very odd because I've been using this version of Itchy for around 6 hours and found no issues with the notifications function.  I can't do anything to debug it on my end, so you'll have to take care of it.

micahlt commented:
With everything working at this point, I'm going to go ahead and merge this.  Since the initial PR, I also: 

- Fixed some navigation issues
- Added more feed events
- Drastically improved profile UI
- Fixed infinite scroll on search
- Improved modularization
- All-around performance improvements
